src/app-client.ts

```diff
@@ -10,3 +13,70 @@ import { ApplicationClient, AppSpecAppDetails } from './types/app-client'
 export function getAppClient(appDetails: AppSpecAppDetails, algod: Algodv2) {
   return new ApplicationClient(appDetails, algod)
 }
+
+/**
+ * Create a new ApplicationClient instance by id
+ * @param app The ARC-0032 application spec as either:
+ *  * Parsed JSON `AppSpec`
+ *  * Raw JSON string
+ * @param id The id of an existing app to call using this client, or 0 if the app hasn't been created yet
+ * @param algod An algod instance
+ * @param sender Default sender to use for transactions issued by this application client
+ * @param params Default suggested params object to use
+ * @param deployTimeParams Optionally provide any deploy-time parameters to replace in the TEAL code; if specified here will get used in calls to `deploy`, `create` and `update` unless overridden in those calls
+ * @param name The optional name to use to mark the app when deploying `ApplicationClient.deploy` (default: uses the name in the ABI contract)
+ *
+ * @returns The application client
+ */
+export function getAppClientById(
+  app: AppSpec | string,
```

Comment:
For this many args I'd suggest an object rather than ordered params.

Also if you do that then the algokit utils convention is to have the `algod` as a separate parameter after the object (check out the other functions to see what I mean)